src/ir/comp.rs

```diff
@@ -97,14 +103,18 @@ pub struct Component {
     /// Externally facing interface information, used to preserve interface in compilation.
     /// Must be `Some` for toplevel components and externals.
     pub src_info: Option<InterfaceSrc>,
+    /// The file that this component came from.
+    /// Must be `Some` for externals
+    pub filename: Option<String>,
```

Comment:
This information is contained in the context. We probably only need to know whether something is an external component or not which can be exposed as a method on the context.

Reply:
The reason I did this I think was because we need to construct a new context, and to do this we need to know what file the underlying component came from so we know the mapping for the new component. I could search through all the entries in the old context's `ext_map` until I find it, but I thought it would be easier to just store that information per component. Let me know which I should do?

Reply:
We should keep that information in the `Ctx` and expose a method to get it. This information is probably not going to be used by any pass that acts upon the component and it's only meaningful for externals.

Reply:
But yeah it does make sense to keep the information in `Ctx`